sdk/keyvault/azure-keyvault-keys/tests/test_key_client.py

```diff
@@ -185,6 +186,9 @@ def _to_bytes(hex):
     def test_key_crud_operations(self, client, is_hsm, **kwargs):
         self.assertIsNotNone(client)
 
+        if (self.is_live and os.environ["KEYVAULT_SKU"] != "premium"):
```

Comment:
This test doesn't require a premium vault.

Reply:
@chlowell I remove this code by your comments. But it failed in pipeline in China cloud. Error message `Hardware key operation not allowed on *** vault`. For more details please check [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1201262&view=logs&j=29c9017d-5931-5b87-96af-c798df6e68e8&t=97b61c76-f672-523b-e188-289102a5aa7d). 
Besides, I skip this test is based on the JS issue https://github.com/Azure/azure-sdk-for-js/issues/18267 and PR https://github.com/Azure/azure-sdk-for-js/pull/18350.
So, I think maybe we should skip this test when `KEYVAULT_SKU != "premium"`. Any ideas?

Reply:
Hi @v-xuto, this actually is coming from a bug: when we [create a key](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-keys/tests/test_key_client.py#L191) at the start of this test, we always create a hardware-protected key (`hardware_protected=True`) instead of doing so just for Managed HSM testing (`hardware_protected=is_hsm`). The same is true for the async version of this test, which is why they were failing for non-premium vaults.

Reply:
@mccoyp I have remove this skip code in async version of this test. When do you plan to fix this test? What are your plans? We can only hold this PR first until you fix this test.

Reply:
@v-xuto I think we can fix the test as part of this PR, if that's alright, since we just need to change `hardware_protected=True` to `hardware_protected=is_hsm` [here](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-keys/tests/test_key_client.py#L191) and [here](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-keys/tests/test_keys_async.py#L186).

Reply:
@mccoyp Thanks for your help. After I have updated by your comments, this issue about `test_key_crud_operations` has been resolved. But there is a new problem about `test_key_rotation_policy` in `public` cloud when I run it in a new pipeline. For more details please check [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1212260&view=logs&j=16f93b6d-ccf1-53e8-60ee-bc5b8717f42a&t=67369ee1-cdad-587c-15a2-c24641cb12fa).
Error message: `azure.core.exceptions.HttpResponseError: (Forbidden) The user, group or application 'appid=***;oid=***;iss=https://sts.windows.net/***/' does not have keys setrotationpolicy permission on key vault 't7d1580f70e294645;location=***'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287`
About this error, do you have any ideas?

Reply:
@v-xuto Sorry for the delay! I was out of office for a bit.

That error is coming from new permissions that are necessary for key release tests -- I just updated our tests in [this PR](https://github.com/Azure/azure-sdk-for-python/pull/21940) to adjust to the new requirements. If you rebase this PR with `main` then the error message should go away.

Reply:
@mccoyp  Thank you for your PR, this PR fixed this error. But I found a new error `Failed to build cryptography.` in `ubuntu2004_pypy3`. I found that the pipeline that your PR runs also has this [problem](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1220316&view=results). Don't we need to care? If necessary, how do we solve it?

Reply:
@v-xuto Thank you for confirming! That `cryptography` issue is a known issue that popped up over the last week, yes. It should have been fixed with [this PR](https://github.com/Azure/azure-sdk-for-python/pull/21960), so re-running pipelines should resolve the issue.

Also, pipelines should pass for this PR if we update recordings for `test_key_client.py` and `test_keys_async.py`'s `test_key_crud_operations` test. Playback tests are [failing for now](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1221643&view=ms.vss-test-web.build-test-results-tab&runId=26234412&resultId=100316&paneView=debug) because recordings for vault tests were always creating HSM keys. Would you be able to add new recordings after running that test live in sync and async, or would you like me to record and push to this PR?

Reply:
@mccoyp  I rerun a new pipeline about this PR, it failed in `Deploy test resources`. For more details please check [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1224388&view=logs&j=16f93b6d-ccf1-53e8-60ee-bc5b8717f42a&t=6ce31d77-520f-5ca1-bdbb-af62fd4f0b50). Do you have any ideas? Besides, I will add recordings for `test_key_crud_operations` test later.

Reply:
@benbp Do you have any suggestions about `deploy test resources failure`? For more details please check here.

Reply:
@v-xuto the `Deploy test resources` failure was caused by a bug that was fixed in [this PR](https://github.com/Azure/azure-sdk-tools/pull/2372), so a re-run of the pipeline should be able to deploy resources successfully

Reply:
@mccoyp I have add all recordings for `test_key_crud_operations` except for the `mhsm` key vault test. Because `mhsm` test not fail in ci pipeline and we run this live test locally hit the error `(Forbidden) The managed HSM resource is currently in Provisioned state. In this state you can only backup the current security domain or restore the security domain of another managed HSM resource`. Please review the recordings. Please let me know if there is any problem.

Reply:
@v-xuto Thank you for adding the recordings! I'll go ahead and approve, everything looks good to me. Thank you for all the work on this!!